### ISSUE TITLE
[Image] Bundle default avatars as SVG to use as a fallback when custom avatars are not loaded or cached

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#9703]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/docs/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. open app and login
2. Go offline 
3. refresh the app 
4. Go back online 

## Expected Result:
All Avatars should display fall back icons 

## Actual Result:
When on the spotty plane wifi but user's own avatar in NewDot is showing the gray “fallback” avatar that we show when we can’t fetch an avatar. 

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.74-2
**Reproducible in staging?:** Y
**Reproducible in production?:** needs repro
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 
<img width="373" alt="image - 2022-07-05T132519 164" src="https://user-images.githubusercontent.com/43995119/177382962-d064fd17-9794-4e45-9f95-1adfeacce47a.png">
<img width="72" alt="image - 2022-07-05T132515 739" src="https://user-images.githubusercontent.com/43995119/177382966-9c460c15-aebb-4c00-810a-1f90f02b7e4a.png">

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @shawnborton 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1654533813143499

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by roryabraham** [comment]
Grabbing this and making it a weekly

**Comment by roryabraham** [comment]
Not going to get to this issue this week

**Comment by roryabraham** [comment]
No update. Probably will not get to this until next week

**Comment by roryabraham** [comment]
Hoping to submit a PR for this today

**Comment by roryabraham** [comment]
Putting this on HOLD and adding it to [this tracking issue](https://github.com/Expensify/App/issues/10894)

**Comment by puneetlath** [comment]
Assigning over to @mallenexpensify since this is part of https://github.com/Expensify/App/issues/10894

**Comment by mallenexpensify** [comment]
Bumping to weekly cuz it's on Hold

**Comment by mallenexpensify** [comment]
On hold, the tracking issue was recently updated so we should be moving forward soon #10894

**Comment by trjExpensify** [comment]
@Beamanator, I believe this is still on hold. Specifically, it's held on #10792 that Margelo are actively working on right? Will update the title to be more specific than the tracker if so!

**Comment by Beamanator** [comment]
Yep still on hold for that one 👍 & technically even holding on https://github.com/Expensify/App/issues/12603

**Comment by trjExpensify** [comment]
When you say _even_ holding on #12603, is that blocking #10792 or I just cited the wrong issue it's actually held on? 😅

**Comment by Beamanator** [comment]
I mean those two issues you linked are both solving the same problem (image caching) but for different platforms - and I don't think we can work on this one until caching is working on all platforms.

**Comment by trjExpensify** [comment]
Excellent, updated. Thanks!

**Comment by Beamanator** [comment]
@roryabraham Are you interested in taking over this one, mainly implementing the "fallback behavior" which would be the default avatar SVGs (not URLs)? I think you mentioned you're interested in implementing that feature :D

**Comment by cristipaval** [comment]
I'll get to this soon. Working on another issue atm, part of WAQ.

**Comment by trjExpensify** [comment]
Nice, thanks @cristipaval! ❤️ 

I've added the `Internal` label, and added this to the WAQ milestone. `#10792` has merged, so updating the title to reflect the remaining web/desktop caching issue we're blocked on.

**Comment by roryabraham** [comment]
Clarifying here, there are really two tasks:

1. load cached images even offline. This should be on HOLD for https://github.com/Expensify/App/issues/12603
2. Bundle default avatars as svg in the repo, use them as a fallback when custom avatars are not loaded or cached. This does not need to be on HOLD

**Comment by roryabraham** [comment]
Yes, agreement

**Comment by trjExpensify** [comment]
Okay, got it. So my suggestion would be to make this more clear than it currently is in the title and OP. If they aren't dependent on each other, then "part 1" / "part 2" doesn't seem intuitive. Let's call this issue: 

`[Image] Bundle default avatars as SVG to use as a fallback when custom avatars are not loaded or cached.`

Cristi doesn't need to be blocked on getting going on that while we wait on #12603 to merge. After it does, we can confirm caching on web/desktop works as intended for avatar images as well.

**Comment by cristipaval** [comment]
Where can I find the svg files?

**Comment by shawnborton** [comment]
Just wanted to clarify that this is the correct fallback avatar link: https://github.com/Expensify/App/blob/main/assets/images/avatars/fallback-avatar.svg

**Comment by roryabraham** [comment]
Oh, whoops thanks @shawnborton. I didn't link to `main`

**Comment by cristipaval** [comment]
Hey @roryabraham, @Beamanator! Looks like the issue is not reproducible anymore. Looking at the [Avatar.js](https://github.com/Expensify/App/blob/main/src/components/Avatar.js) source code, the fallback avatar is already implemented and points to the desired svg. Please also see below some tests with Storybook (by the way thanks Rory for suggesting me to use this tool, I love it!)
Am I missing anything?

https://user-images.githubusercontent.com/18078393/204826545-d4e2999e-70ae-4fc2-9029-5e2295ae3ab6.mov

**Comment by cristipaval** [comment]
One more thing:
Yes, the fallback icon is an SVG bundled in the App, is not an cloud front url.
See [this](https://github.com/Expensify/App/blob/1527fdcf443713bb058e6e76504006d74f6b1aa8/src/components/Avatar.js#L41) piece of code in the Avatar.js and [this one ](https://github.com/Expensify/App/blob/1527fdcf443713bb058e6e76504006d74f6b1aa8/src/components/Icon/Expensicons.js#L96)in the Expesicons.js

**Comment by cristipaval** [comment]
I had a 1:1 with @Beamanator. He helped me understand the context a bit more. So basically the default avatar is different than fallback avatar.
- The fallback avatar has been showing whenever there is an error while fetching the image from a specific source
- The default avatar is shown when there is no custom avatar uploaded for the account

We want to bundle the default avatars in the App instead of fetching them from the cloud front. This opens up some other questions:
1. Do we still need the fallback avatar if we have the default avatars bundled in the App? We could just show the default avatar whenever a custom avatar fails to download.
2. Georgia added new default avatars in [this PR](https://github.com/Expensify/Web-Static/pull/120/files#diff-7a62b2bee64c9025e6744a574bd0301932870e44be297f0d37810ad52abb5ca4). I guess we want to use these new avatars instead of the old ones. The thing is that these avatars are png files. @shawnborton is there a way to convert them to svg? Looking at [this Slack convo](https://expensify.slack.com/archives/C03TQ48KC/p1669232406744319) my suggestion is to bundle the png files in the App for now and then work on implementing support for svg avatars in a separate PR.
What do you guys think?

**Comment by cristipaval** [comment]
Thanks @shawnborton! I only see png files [here](https://github.com/Expensify/Web-Static/tree/main/images/avatars). COuld you help me find the svg files for the new default avatars?

**Comment by shawnborton** [comment]
CC @grgia as she is working through the default avatars as well - let's make sure both of this issue and the default avatar issue work nicely together?

In terms of the SVGs, they aren't on cloudfront but I can provide them to you in a folder here? But thinking more on this... maybe we do just want one default fallback avatar to use when we can't fetch the defaults?

**Comment by shawnborton** [comment]
Yeah, chatting more with Georgia and Yuwen, I think the title of this issue is misleading and isn't actually reflective of the original issue I reported. The original issue I reported was that I had an avatar uploaded, but then I went offline and the app started showing a fallback offline avatar. It sounds like that's actually going to be fixed here: https://github.com/Expensify/App/pull/12808

So that being said, my vote is to just close this one out. Thoughts?

**Comment by cristipaval** [comment]
I totally agree with you @shawnborton. I think that after we merge https://github.com/Expensify/App/pull/12808 into master, it doesn't make a lot of sense to bundle default avatars into the App given the fact that we're going to cache the avatars for offline usage (as @roryabraham mentioned in the first task [here](https://github.com/Expensify/App/issues/9703#issuecomment-1330275705)). Once we have the avatars cached it will be quite rare when we will see the fallback avatar and technically I don't a specific use case/corner case when the bundled default avatars could improve the ux. I am also leaning towards closing this issue.
cc @Beamanator @trjExpensify @JmillsExpensify

**Comment by cristipaval** [comment]
Yes, this is the same context that I have. Thanks for putting it in order.
Now, given the fact that we're going to cache custom avatars for offline usage, I see 2 approaches:
1. Do nothing and the default avatars will be cached as well as the custom ones. Whenever the App is offline and the cache is corrupt the fallback avatar is shown (which is a rare case I'd say).
2. Bundle default avatars in the App, get rid of fallback avatar and implement a mechanism to display defaults instead of the fallback icon.
I am leaning towards 1 because I don't expect the users to see fallback avatar quite often.

**Comment by trjExpensify** [comment]
>I am leaning towards 1 because I don't expect the users to see fallback avatar quite often.

Doesn't that mean that in all instances of things like starting a chat with someone new while offline via places like `New chat` or sending a money request, we'd be using the fallback avatar until they come back online and it loads for the first time? 

>In this issue I believe the plan is to load these files directly into App so we don't have to load from Cloudfront, meaning we don't need a "fallback" anymore since users will always have access to these assets

I thought this was the point of the revised plan, so we do away with this notion of a "fallback" avatar and instead show the new default avatar in those instances.

**Comment by JmillsExpensify** [comment]
> Doesn't that mean that in all instances of things like starting a chat with someone new while offline via places like New chat or sending a money request, we'd be using the fallback avatar until they come back online and it loads for the first time?

Yeah I think 1 is basically saying that there are (edge) cases where a fallback avatar is useful. However, we could also say that given that this case should be so rare, we don't do anything about and remove the concept of the fallback avatar entirely.

**Comment by cristipaval** [comment]
> Doesn't that mean that in all instances of things like starting a chat with someone new while offline via places like New chat or sending a money request, we'd be using the fallback avatar until they come back online and it loads for the first time?
Oh, this is indeed a concrete use case which I missed and I think it's beautiful to have default avatars.

Ok, so I'll just continue my work on the issue. We get rid of the fallback icon and we bundle them in the App to be used whenever no custom avatar is available to load.

**Comment by shawnborton** [comment]
@cristipaval can you coordinate with @grgia on this one? I know she started [this PR here](https://github.com/Expensify/Web-Expensify/pull/35715) so just want to make sure we aren't doing overlapping work

**Comment by cristipaval** [comment]
> @cristipaval can you coordinate with @grgia on this one? I know she started [this PR here](https://github.com/Expensify/Web-Expensify/pull/35715) so just want to make sure we aren't doing overlapping work

Sure, thanks for linking me to her PR!

**Comment by Beamanator** [comment]
Also quite tiny note @JmillsExpensify :

> Yeah I think 1 is basically saying that there are (edge) cases where a fallback avatar is useful. However, we could also say that given that this case should be so rare, we don't do anything about and remove the concept of the fallback avatar entirely.

To be clear, we need the fallback avatar _iff_ the default avatars are held in Cloudfront (as they are now), because there's flows where we might open a chat offline (brought up above) and probably more where we don't have internet access & are trying to pull the custom avatar or default avatar from the interwebs. But it seems we're all in agreement that it makes the most sense to make the default avatars even available offline so we will never need the Fallback avatar (a.k.a. the defaults become the fallback)

**Comment by trjExpensify** [comment]
Agree with the outcome here! 👍

**Comment by shawnborton** [comment]
Here are the SVGs we'd need if we go that route: [Random User Icon.zip](https://github.com/Expensify/App/files/10133919/Random.User.Icon.zip)

**Comment by JmillsExpensify** [comment]
Yeah love this! Great discussion.

**Comment by cristipaval** [comment]
> @cristipaval can you coordinate with @grgia on this one? I know she started [this PR here](https://github.com/Expensify/Web-Expensify/pull/35715) so just want to make sure we aren't doing overlapping work

I had a 1:1 with @grgia and we realised that we were doing overlapping work. She is ahead of me and we decided that in order to implement the solution that we decided here it's easier to update her PRs ([this one](https://github.com/Expensify/Web-Expensify/pull/35715) on WebE and [this one](https://github.com/Expensify/App/pull/13216) on the App). I'll assist her if she needs help.

**Comment by trjExpensify** [comment]
Given that this is being taken care of in the WIP PRs to update the default avatars, how do we feel about closing this one out at this point then as there isn't any further action to take here?

**Comment by trjExpensify** [comment]
👋 discussed this one [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1670271704874249), let's close it in favour of the issue Georgia is working on [here](https://github.com/Expensify/App/issues/12259) to update the default avatars.
